File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        int mode = 0;
        String result = "";
        for(int idx = 0 ; idx < code.length() ; idx++){
            if(code.charAt(idx) == '1'){
                if(mode == 0) mode = 1;
                else mode = 0;
                continue;
            }
            switch (mode) {
                case 0 : 
                    if(idx % 2 == 0) result += code.charAt(idx);
                    break;
                case 1 : 
                    if(idx % 2 == 1) result += code.charAt(idx);
                    break;
            }
        }
        return result.isEmpty()?"EMPTY":result;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ""
    mode = 0
    
    for i in range(len(code)):
        if code[i] == "1":
            mode = 1 if mode == 0 else 0
        
        if mode == 0:
            if code[i] != "1" and i%2==0:
                answer += code[i]
        else:
            if code[i] != "1" and i%2==1:
                answer += code[i]
            
    return answer if answer else "EMPTY"